File: app/src/main/java/com/nfcalarmclock/permission/NacOptionalPermissionPreference.java
package com.nfcalarmclock.permission;

import android.content.Context;
import android.content.res.Resources;
import android.util.AttributeSet;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.preference.Preference;
import androidx.preference.PreferenceViewHolder;
import com.nfcalarmclock.R;

/**
 * A preference that is used to display optional permissions.
 */
public class NacOptionalPermissionPreference
	extends Preference
{

	/**
	 */
	public NacOptionalPermissionPreference(Context context)
	{
		super(context);
		this.init();
	}

	/**
	 */
	public NacOptionalPermissionPreference(Context context, AttributeSet attrs)
	{
		super(context, attrs);
		this.init();
	}

	/**
	 */
	public NacOptionalPermissionPreference(Context context, AttributeSet attrs, int style)
	{
		super(context, attrs, style);
		this.init();
	}

	/**
	 * Initialize the preference.
	 */
	private void init()
	{
		setLayoutResource(R.layout.nac_preference_permission);
	}

	/**
	 */
	@Override
	public void onBindViewHolder(@NonNull PreferenceViewHolder holder)
	{
		super.onBindViewHolder(holder);

		Context context = getContext();
		Resources res = context.getResources();
		TextView permissionType = (TextView) holder.findViewById(R.id.permission_type);

		// Get the text and color
		String text = res.getString(R.string.message_optional).toLowerCase();
		int color = res.getColor(R.color.green);

		// Setup the textview
		permissionType.setText(text);
		permissionType.setTextColor(color);
	}

}

File: app/src/main/kotlin/com/nfcalarmclock/permission/NacOptionalPermissionPreference.kt
package com.nfcalarmclock.permission

import android.content.Context
import android.os.Build
import android.util.AttributeSet
import android.widget.TextView
import androidx.preference.Preference
import androidx.preference.PreferenceViewHolder
import com.nfcalarmclock.R
import java.util.Locale

/**
 * A preference that is used to display optional permissions.
 */
class NacOptionalPermissionPreference
	: Preference
{

	/**
	 * Constructor.
	 */
	constructor(context: Context) : super(context)
	{
	}

	/**
	 * Constructor.
	 */
	constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)
	{
	}

	/**
	 * Constructor.
	 */
	constructor(context: Context, attrs: AttributeSet?, style: Int) : super(context, attrs, style)
	{
	}

	/**
	 * Constructor.
	 */
	init
	{
		layoutResource = R.layout.nac_preference_permission
	}

	/**
	 * Called when the view holder is bound.
	 */
	override fun onBindViewHolder(holder: PreferenceViewHolder)
	{
		// Super
		super.onBindViewHolder(holder)

		// Get the textview
		val permissionType = holder.findViewById(R.id.permission_type) as TextView

		// Setup the textview
		setPermissionText(permissionType)
		setPermissionTextColor(permissionType)
	}

	/**
	 * Set the permission text.
	 */
	private fun setPermissionText(textView: TextView)
	{
		// Get the message
		val locale = Locale.getDefault()
		val message = context.resources.getString(R.string.message_optional)

		// Set the text
		textView.text = message.lowercase(locale)
	}

	/**
	 * Set the color of the permission text.
	 */
	@Suppress("deprecation")
	private fun setPermissionTextColor(textView: TextView)
	{
		// Get the color based on the API
		val color = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M)
		{
			context.getColor(R.color.green)
		}
		else
		{
			context.resources.getColor(R.color.green)
		}

		// Set the color
		textView.setTextColor(color)
	}

}